### PR TITLE
fix(agent): dagent registry auth

### PR DESCRIPTION
**Pull request by robot9706**
Dagent is unable to check image manifests due to incompatible registry auth formats.

### CODE REVIEW

**Review by polaroi8d — APPROVED** [review]
LGTM (informal)

File: golang/internal/helper/image/image.go [hunk 1]
```diff
@@ -246,7 +247,11 @@ func checkRemote(ctx context.Context, check remoteCheck) (err error) {
 	craneOpts := []crane.Option{}
 
 	if check.encodedAuth != "" {
-		craneOpts = append(craneOpts, crane.WithAuth(authn.FromConfig(authn.AuthConfig{Auth: check.encodedAuth})))
+		basicAuth, convertError := authConfigToBasicAuth(check.encodedAuth)
+		if err != nil {
```
Comment (nandor-magyar) [inline golang/internal/helper/image/image.go:251]: I assume you wanted to check `convertError`.

**Review by nandor-magyar — CHANGES_REQUESTED** [review]

File: golang/internal/helper/image/image.go (hunk 1 above)
Comment (robot9706) [reply golang/internal/helper/image/image.go]: Oops, thanks.

**Review by nandor-magyar — APPROVED** [review]
LGTM

**Pull request merged**